src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/insights/BitbucketServerCodeInsightsPullRequestDecorator.java

```diff
@@ -0,0 +1,182 @@
+        reportData.add(securityReport(rules.get(RuleType.VULNERABILITY), rules.get(RuleType.SECURITY_HOTSPOT)));
+        reportData.add(new ReportData("Duplication", new DataValue.Percentage(newDuplication(analysisDetails))));
+        reportData.add(maintainabilityReport(rules.get(RuleType.CODE_SMELL)));
+        reportData.add(new ReportData("Analysis details", new DataValue.Link("Go to SonarQube", analysisDetails.getDashboardUrl())));
+
+        return new CreateReportRequest(reportData,
+                reportDescription(analysisDetails),
+                "SonarQube",
+                "SonarQube",
+                analysisDetails.getAnalysisDate().toInstant(),
+                analysisDetails.getDashboardUrl(),
+                "https://www.sonarqube.org/favicon-152.png",
+                asInsightStatus(analysisDetails.getQualityGateStatus()));
+    }
+
+    private void updateAnnotations(String project, String repo, AnalysisDetails analysisDetails) throws IOException {
+        client.deleteAnnotations(project, repo, analysisDetails.getCommitSha());
+
+        Set<Annotation> annotations = analysisDetails.getPostAnalysisIssueVisitor().getIssues().stream()
+                .filter(i -> i.getComponent().getReportAttributes().getScmPath().isPresent())
+                .filter(i -> i.getComponent().getType() == Component.Type.FILE)
+                .filter(i -> OPEN_ISSUE_STATUSES.contains(i.getIssue().status()))
+                .map(componentIssue -> {
+                    String path = componentIssue.getComponent().getReportAttributes().getScmPath().get();
+                    return new Annotation(componentIssue.getIssue().key(),
+                            Optional.ofNullable(componentIssue.getIssue().getLine()).orElse(0),
+                            analysisDetails.getIssueUrl(componentIssue.getIssue().key()),
+                            componentIssue.getIssue().getMessage(),
+                            path,
+                            toBitbucketSeverity(componentIssue.getIssue().severity()),
+                            toBitbucketType(componentIssue.getIssue().type().name()));
+                }).collect(Collectors.toSet());
+
+        client.createAnnotations(project, repo, analysisDetails.getCommitSha(), new CreateAnnotationsRequest(annotations));
+    }
+
+    private String asInsightStatus(QualityGate.Status status) {
+        return QualityGate.Status.ERROR.equals(status) ? "FAIL" : "PASS";
+    }
+
+    private String toBitbucketSeverity(String severity) {
+        if (severity == null) {
+            return "LOW";
+        }
+        switch (severity) {
+            case "BLOCKER":
+            case "CRITICAL":
+                return "HIGH";
+            case "MAJOR":
+                return "MEDIUM";
+            default:
+                return "LOW";
+        }
+    }
+
+    private String toBitbucketType(String sonarqubeType) {
+        switch (sonarqubeType) {
+            case "SECURITY_HOTSPOT":
```

Comment:
we could use RuleType eg. RuleType.SECURITY_HOTSPOT.name()

Reply:
Fixed